packages/angular-cli/plugins/karma.js

```diff
@@ -17,6 +19,25 @@ const init = (config) => {
     progress: config.angularCli.progress
   }
 
+  // add assets
+  if (appConfig.assets) {
+    const assets = typeof appConfig.assets === 'string' ? [appConfig.assets] : appConfig.assets;
+    config.proxies = config.proxies || {};
+    assets.forEach(asset => {
+      const fullAssetPath = path.join(config.basePath, appConfig.root, asset);
+      const isDirectory = fs.lstatSync(fullAssetPath).isDirectory();
+      const filePattern = isDirectory ? fullAssetPath + '/**' : fullAssetPath;
+      const proxyPath = isDirectory ? asset + '/' : asset;
+      config.files.push({
+        pattern: filePattern,
+        included: false,
+        served: true,
+        watched: true
+      });
+      config.proxies['/' + proxyPath] = '/base/' + appConfig.root + '/' + proxyPath;
```

Comment:
Yes that makes sense to me. I remember a similar situation I ran into another project. Without the proxy, fetch would not work correctly.